gpflow/models/het_gpr.py

```diff
@@ -0,0 +1,108 @@
+import tensorflow as tf
+from typing import Optional
+
+from gpflow import posteriors
+from ..kernels import Kernel
+from ..logdensities import multivariate_normal
+from ..mean_functions import MeanFunction
+from ..models.gpr import GPR_with_posterior
+from ..models.training_mixins import RegressionData, InputData
+from ..types import MeanAndVariance
+from ..utilities import add_linear_noise_cov
+
+
+class het_GPR(GPR_with_posterior):
+    """ While the vanilla GPR enforces a constant noise variance across the input space, here we allow the
+    noise amplitude to vary linearly (and hence the noise variance to change quadratically) across the input space.
```

Comment:
Is it true that the noise amplitude varies linearly with this model? Even with the `PseudoPoissonLikelihood`?